File: functions/src/firestoreUtils.ts

import {getFirestore} from 'firebase-admin/firestore';
import * as logger from 'firebase-functions/logger';

export async function getCodeExists(code: string) {
  console.log('in getCodeExists', code);
  let existingWithSameCode = getFirestore().collectionGroup('lists')
    .where('shareCodeForViewer', '==', code);
  let snapshot = await existingWithSameCode.count().get();
  logger.log('count', snapshot.data().count);
  if (snapshot.data().count > 0) {
    return true;
  }

  existingWithSameCode = getFirestore().collectionGroup('lists')
    .where('shareCodeForEditor', '==', code);
  snapshot = await existingWithSameCode.count().get();
  logger.log('count', snapshot.data().count);
  if (snapshot.data().count > 0) {
    return true;
  }

  return false;
}

File: functions/firestore_utils.py

import firebase_admin
from firebase_admin import firestore

default_app = firebase_admin.initialize_app()

def check_code_exists(code):
    db = firestore.client()
    
    lists_ref = db.collection_group('lists')
    
    query1 = lists_ref.where('shareCodeForViewer', '==', code).limit(1).get()
    query2 = lists_ref.where('shareCodeForEditor', '==', code).limit(1).get()
    
    if len(query1) > 0 or len(query2) > 0:
        return True
    else:
        return False
